src/engine/sched.c

```diff
@@ -1086,16 +1145,75 @@ policy_fifo_enqueue(struct dss_xstream *dx, struct sched_request *req,
 		    void *prio_data)
 {
 	struct sched_info	*info = &dx->dx_sched_info;
+	struct sched_req_attr	*attr = &req->sr_attr;
+	int			 rc;
 
+	D_ASSERT(attr->sra_type < SCHED_REQ_TYPE_MAX);
+	/*
+	 * If @sra_enqueue_ts is not zero, this is a resent RPC, it will
+	 * be inserted to sorted heap. if it failed to insert because of
+	 * out of memory, use fifo list instead.
+	 */
+	if (attr->sra_enqueue_ts) {
+		rc = d_binheap_insert(&info->si_heap, &req->sr_node);
+		if (rc)
+			goto out;
+		return;
+	}
```

Comment:
This retry request tracking will be per policy instead of common code shared by different polices? I don't against this, just ask to confirm.

Reply:
Yes.